cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -84,8 +76,19 @@ public DatasetServiceClient load(Id.Namespace namespace) throws Exception {
         return new DatasetServiceClient(discoveryClient, namespace);
       }
     });
-    this.registryFactory = registryFactory;
-    this.typeLoader = typeLoader;
+    this.instances = new AbstractDatasetProvider(registryFactory, typeLoader) {
+      @Override
+      public DatasetMeta getMeta(Id.DatasetInstance instance) throws Exception {
+        return RemoteDatasetFramework.this.clientCache.getUnchecked(instance.getNamespace())
+          .getInstance(instance.getId());
+      }
+
+      @Override
+      public void createIfNotExists(Id.DatasetInstance instance, String type,
+                                    DatasetProperties creationProps) throws Exception {
```

Comment:
Oops. Everything worked fine because it's not used, but this should be implemented.